modules/standard/Socket.chpl

```diff
@@ -0,0 +1,1282 @@
+  Waits for a new connection based on `timeout` and
+  returns a new :type:`tcpConn` if successful. Default
+  time to wait for a new connection is indefinite.
+
+  .. code-block:: Chapel
+
+    const client = server.accept()
+
+  :arg timeout: time to wait for new connection.
+  :type timeval: :type:`~Sys.timeval`
+  :return: accepted connection.
+  :rtype: `tcpConn`
+  :throws Error: Upon timeout completion without
+                  any new connection
+*/
+proc tcpListener.accept(in timeout: timeval = new timeval(-1,0)):tcpConn throws {
+  var client_addr:sys_sockaddr_t = new sys_sockaddr_t();
+  var fdOut:fd_t;
+  var err_out:err_t = 0;
+  // try accept
+  err_out = sys_accept(socketFd, client_addr, fdOut);
+  // if error is not about blocking throw error
+  if err_out != 0 && err_out != EAGAIN && err_out != EWOULDBLOCK {
+    throw SystemError.fromSyserr(err_out, "accept() failed");
+  }
+  // successfully return file
+  if err_out == 0 {
+    return openfd(fdOut):tcpConn;
+  }
+  var localSync$: sync c_short;
+  // create event pending state
+  var internalEvent = event_new(event_loop_base, this.socketFd, EV_READ | EV_TIMEOUT, c_ptrTo(syncRWTCallback), c_ptrTo(localSync$):c_void_ptr);
+  defer {
+    // cleanup
+    event_free(internalEvent);
+  }
+  var t: Timer;
+  t.start();
+  // make event active
+  if timeout.tv_sec == -1 {
+    err_out = event_add(internalEvent, nil);
+  }
+  else {
+    err_out = event_add(internalEvent, c_ptrTo(timeout));
+  }
+  if err_out != 0 {
+    throw new Error("accept() failed");
+  }
+  // return value
+  var retval = localSync$.readFE();
+  // stop timer
+  t.stop();
+  // if error was timeout throw error
+  if retval & EV_TIMEOUT != 0 {
+    throw SystemError.fromSyserr(ETIMEDOUT, "accept() timed out");
+  }
+  var elapsedTime = t.elapsed(TimeUnits.microseconds):c_long;
+  // try accept again
+  err_out = sys_accept(socketFd, client_addr, fdOut);
+  if err_out != 0 {
+    // error was not about blocking wait so throw it
+    if err_out != EAGAIN && err_out != EWOULDBLOCK {
+      throw SystemError.fromSyserr(err_out, "accept() failed");
+    }
+    // no indefinitely blocking wait
+    if timeout.tv_sec == -1 {
+      var totalTimeout = timeout.tv_sec*1000000 + timeout.tv_usec;
+      // timer didn't elapsed
+      if totalTimeout > t.elapsed(TimeUnits.microseconds) {
+        const remainingMicroSeconds = ((totalTimeout - elapsedTime)%1000000);
+        const remainingSeconds = ((totalTimeout - elapsedTime)/1000000);
+        return this.accept(new timeval(remainingSeconds, remainingMicroSeconds));
+      }
+      throw SystemError.fromSyserr(err_out, "accept() failed");
+    }
+    return this.accept();
+  }
+  return openfd(fdOut):tcpConn;
+}
+
+/*
+  Close the file descriptor
+*/
+proc tcpListener.close() {
+  sys_close(this.socketFd);
+}
+
+/*
+  Returns the address on which socket is
+  listening on and bound to.
+
+  :return: bound address
+  :rtype: `ipAddr`
+*/
+proc tcpListener.addr throws {
```

Comment:
It seems really weird that this can throw. Can you describe the situation a user would write that would cause it to throw? If it is not possible for a user to run in to it, we don't need this to throw.

Reply:
my reasons for using this are the same as https://github.com/chapel-lang/chapel/pull/17960#discussion_r716154610 when user will use an uninitialized `tcpListener`

Reply:
Right. So we can make a `tcpListener.init()` that creates an acceptable default and ensure that the necessary properties are there for any other `init` for tcpListener.